### PR TITLE
Fixing and integration testing the RDS encoder

### PR DESCRIPTION
**Pull request by patrickdbakke**

### CODE REVIEW

File: announcer.go [hunk 1]
```diff
@@ -39,21 +38,23 @@ type playlist struct {
...
-		cli.StringFlag{Name:"chirp", Value:"https://chirpradio.appspot.com/api/current_playlist", Usage:"URL of the CHIRP current_playlist API endpoint"},
-		cli.BoolFlag{Name:"verbose", Usage:"Run in Verbose mode."},
-		cli.BoolFlag{Name:"test", Usage:"Run in test mode. Sends nothing to Prostream"},
-		cli.BoolFlag{Name:"runOnce", Usage:"Run once and then quit"},
-		cli.StringFlag{Name:"rds", Value:"", Usage:"IP address or hostname of the RDS Encoder"},
-		cli.IntFlag{Name:"pdsPort", Value:23, Usage:"Port used by the RDS Encoder"},
-	}
-	app.Action = func(c *cli.Context) {
-		parseAndRun(c)
+	app := &cli.App{
```
Comment (patrickdbakke) [inline announcer.go:41]: Change here represent an update in the `github.com/urfave/cli` library, and the "dps" "rds" typo.

File: announcer.go [hunk 2]
```diff
@@ -119,13 +120,14 @@ func grabAndSendData(prostream string, prostreamPort int, rds string, rdsPort in
...
 
+	niceUrl := makeNiceUrl(chirpApi)
+	
 	writemsg(fmt.Sprintf("About to GET %s", niceUrl), verbose)
-
 	resp, err := http.Get(niceUrl)
+
 	if err != nil {
-		log.Printf("Error connecting to the CHIRP api: , %s\n", err.Error())
+		fmt.Printf("Error connecting to the CHIRP api: , %s\n", err.Error())
```
Comment (patrickdbakke) [inline announcer.go:130]: I changed all the `log`s to `fmt` so they go to STDOUT and can get unit tested.

Comment (lund5000) [reply announcer.go:130]: I am not a Go expert, but a little research tells me that `log` is safe with concurrent go routines (which `grabAndSendData()` uses) where `fmt` is not. `log` can also be configured to write to STDOUT: `log.SetOutput(os.Stdout)`

That's more about learning, though. It seems like the risk of using `fmt` is where you have so many concurrent writes that the OS starts to interleave them, and I don't think we'll be anywhere near that limit here nor have a problem separating the messages if necessary.

File: announcer.go [hunk 3]
```diff
@@ -224,11 +230,12 @@ func makeRDSMessage(currentTrack track) string {
 	if len(baseMessage) + len(stationID) < 128 {
 		totalMessage = baseMessage + stationID
 	}
+	writemsg(fmt.Sprintf("Sending to RDS: %+s", "DPS="+totalMessage + "\n"), verbose);
```
Comment (lund5000) [inline announcer.go:233]: Nit: I don't think that `makeRDSMessage()` should be logging a message about _sending_ that message.

**Review by lund5000 — APPROVED** [review]
The changes to the code look good, though I don't approve of adding JavaScript tests to a Go project in the long run. I started working on replacing them with Go tests as a little learning project, but that's no reason to hold up trying to get the RDS messages sent in production.

**Comment by patrickdbakke** [comment]
> The changes to the code look good, though I don't approve of adding JavaScript tests to a Go project in the long run. I started working on replacing them with Go tests as a little learning project, but that's no reason to hold up trying to get the RDS messages sent in production.

Fair. Since they're built as integration tests, not unit tests, and build real connections to UDP and TCP servers, I'm less worried about it. 

This should free us up to use any language we want in the future. (ex: likely able to find more maintainers if we convert to JS or ruby).

File: announcer.go (hunk 2 above)
Comment (patrickdbakke) [reply announcer.go:130]: Good call. I didn't dig to hard at go. This is an easy change to make.

**Pull request merged**